hypothesis-python/tests/common/strategies.py

```diff
@@ -48,3 +50,27 @@ def do_draw(self, data):
                 self.accepted.add(x)
                 return True
         return False
+
+
+def build_intervals(ls):
+    ls.sort()
+    result = []
+    for u, l in ls:
+        v = u + l
+        if result:
+            a, b = result[-1]
+            if u <= b + 1:
+                result[-1] = (a, v)
+                continue
+        result.append((u, v))
+    return result
+
+
+def IntervalLists(min_size=0):
+    return st.lists(
+        st.tuples(st.integers(0, 200), st.integers(0, 20)),
+        min_size=min_size,
+    ).map(build_intervals)
+
+
+Intervals = st.builds(IntervalSet, IntervalLists())
```

Comment:
pulled out to `common` for usage in `draw_string_kwargs`.

Reply:
How about `st.lists(st.integers(0, sys.max_unicode), unique=True).map(sorted).map(<itertools.batched ish>).map(IntervalSet)`?  Certain to be non-overlapping, which seems a little more efficient /  avoids some redundancy.

(very happy to hear "out of scope, I'm just moving the existing code" of course)

Reply:
I started working on this, but fell into a performance pit. We convert `IntervalSet` to a string of characters to pass to `st.text()`, which are then converted back into an `IntervalSet` internally (and the repeated union calls in that code are a performance red flag to me, though I haven't profiled). 

I would rather use `data.draw_string` directly, which accepts an `IntervalSet`...but I ran into issues with `@given(st.data())` on `draw_string_kwargs` when I tried this previously:

```
FAILED hypothesis-python/tests/conjecture/test_forced.py::test_forced_string[False-True] - hypothesis.errors.InvalidArgument: Cannot apply @given to a function with defaults.
```